### PR TITLE
Add Monitoring Provisioning #7

### PR DESCRIPTION
**Pull request by ossarga**
- Update install command to install and configure
  prometheus and grafana on monitoring instance
- Update start to setup a grafana dashboard
- Add option to start to start all services

### CODE REVIEW

File: src/main/kotlin/com/thelastpickle/tlpcluster/configuration/TFState.kt [hunk 1]
```diff
@@ -10,12 +10,6 @@ import java.util.LinkedHashMap
 
 typealias HostList = List<Host>
 
-/**
- * Returns a list of IPs ready for PSSH's -H format, exported as an ENV
- */
-fun HostList.toEnv() = "PSSH_HOSTNAMES=" + map { "-H ${it.public}" }.joinToString(" ")
```
Comment (ossarga) [inline src/main/kotlin/com/thelastpickle/tlpcluster/configuration/TFState.kt]: Took this out as I changed `Pssh` to just assign the ip address to a viable. I've added the `-H` into the script files that required it. The reason is the script that created the dashboard needed the host ips without the `-H` in front of it.

File: src/main/kotlin/com/thelastpickle/tlpcluster/commands/Install.kt [hunk 2]
```diff
@@ -26,24 +27,41 @@ class Install(val context: Context) : ICommand {
...
-                )
-            },
-            { println("Failed to copy provisioning resources to all nodes. " + it.message + retryMessage) }
-        )
+        val parallelSsh = Pssh(context, sshKeyPath)
+        val retryMessage = "This could be a transient error; try rerunning the install command."
+        var installSuccessful = true
+
+        ServerType
+            .values()
```
Comment (rustyrazorblade) [inline src/main/kotlin/com/thelastpickle/tlpcluster/commands/Install.kt]: This is clever, I don't know if I would have thought of it.

File: src/main/kotlin/com/thelastpickle/tlpcluster/commands/Install.kt [hunk 3]
```diff
@@ -26,24 +27,41 @@ class Install(val context: Context) : ICommand {
...
+            .filter { context.tfstate.getHosts(it).count() > 0 }
```
Comment (rustyrazorblade) [inline src/main/kotlin/com/thelastpickle/tlpcluster/commands/Install.kt]: Minor nit, I don't think you need the filter here

File: src/main/kotlin/com/thelastpickle/tlpcluster/commands/Install.kt [hunk 4]
```diff
@@ -26,24 +27,41 @@ class Install(val context: Context) : ICommand {
...
+            .forEach {
```
Comment (rustyrazorblade) [inline src/main/kotlin/com/thelastpickle/tlpcluster/commands/Install.kt]: Any thoughts on moving the inside of this into a function that takes a server type, just to break it up a little?  I think it'll help it be more testable when i start doing that docker testing we talked a little bit about.  

I wouldn't hold up the merge for this ^^, just a nice to have / musing.

File: src/main/kotlin/com/thelastpickle/tlpcluster/commands/Start.kt [hunk 5]
```diff
@@ -4,24 +4,76 @@ import com.beust.jcommander.Parameter
...
 
     override fun execute() {
-        check(sshKey.isNotBlank())
+        val sshKeyPath = context.userConfig.sshKeyPath
+
+        check(sshKeyPath.isNotBlank())
 
-        if (!File(sshKey).exists()) {
-            println("Unable to find SSH key $sshKey.")
+        if (!File(sshKeyPath).exists()) {
```
Comment (rustyrazorblade) [inline src/main/kotlin/com/thelastpickle/tlpcluster/commands/Start.kt]: Good catch.  As a follow up we'll probably want to do this check when we generate the yaml.

File: src/main/kotlin/com/thelastpickle/tlpcluster/configuration/TFState.kt (hunk 1 above)
Comment (rustyrazorblade) [inline src/main/kotlin/com/thelastpickle/tlpcluster/configuration/TFState.kt]: Is this removed b/c it's only used in one spot (I think)?

Comment (ossarga) [reply src/main/kotlin/com/thelastpickle/tlpcluster/configuration/TFState.kt]: Yes. It was only used in one spot.

File: src/main/kotlin/com/thelastpickle/tlpcluster/commands/Start.kt (hunk 5 above)
Comment (ossarga) [reply src/main/kotlin/com/thelastpickle/tlpcluster/commands/Start.kt]: Agreed

File: src/main/kotlin/com/thelastpickle/tlpcluster/commands/Install.kt (hunk 3 above)
Comment (ossarga) [reply src/main/kotlin/com/thelastpickle/tlpcluster/commands/Install.kt]: As discussed we should probably leave this in for now.

File: src/main/kotlin/com/thelastpickle/tlpcluster/commands/Install.kt (hunk 2 above)
Comment (ossarga) [reply src/main/kotlin/com/thelastpickle/tlpcluster/commands/Install.kt]: Thanks 👍

File: src/main/kotlin/com/thelastpickle/tlpcluster/commands/Install.kt (hunk 4 above)
Comment (ossarga) [reply src/main/kotlin/com/thelastpickle/tlpcluster/commands/Install.kt]: Done

**Comment by ossarga** [comment]
Merging as discussed with @rustyrazorblade

**Pull request merged**